iib/web/models.py

```
@@ -1435,3 +1436,101 @@ def validate_registry_auths(registry_auths):
+    from_index = db.relationship('Image', foreign_keys=[from_index_id], uselist=False)
+    binary_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    binary_image = db.relationship('Image', foreign_keys=[binary_image_id], uselist=False)
+    _labels = db.Column('labels', db.Text, nullable=True)
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['create_empty_index'].value
+    }
+
+    @property
+    def labels(self):
+        """Return the Python representation of the JSON labels."""
+        return json.loads(self._labels) if self._labels else None
+
+    @labels.setter
+    def labels(self, labels):
+        """
+        Set the labels column to the input labels as a JSON string.
+
+        If ``None`` is provided, it will be simply set to ``None`` and not be converted to JSON.
+
+        :param dict labels: the dictionary of the labels or ``None``
+        """
+        self._labels = json.dumps(labels, sort_keys=True) if labels is not None else None
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the create-empty-index API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        if request_kwargs.get('from_index') is None:
+            raise ValidationError('"from_index" must be a non-empty string')
+
+        labels = request_kwargs.pop('labels', {})
+        if not isinstance(labels, dict):
+            raise ValidationError('The value of "labels" must be a JSON object')
+
+        for arg in (
+            'add_arches',
+            'overwrite_from_index',
+            'overwrite_from_index_token',
+            'distribution_scope',
+        ):
+            if request_kwargs.get(arg) is not None:
```

`if arg in request_kwargs:`? We want to throw an error irrespective of the value right.